src/viam/examples/modules/complex/base/impl.cpp

```diff
@@ -63,18 +64,26 @@ bool MyBase::is_moving() {
     return left_->is_moving() || right_->is_moving();
 }
 
-grpc::StatusCode MyBase::stop(const AttributeMap& extra) {
-    auto left_stop = left_->stop(extra);
-    auto right_stop = right_->stop(extra);
+void MyBase::stop(const AttributeMap& extra) {
+    std::exception* e;
 
-    // Return first of any non-ok error code received from motors.
-    if (left_stop != grpc::StatusCode::OK) {
-        return left_stop;
+    // make sure we try to stop both motors, even if the first fails.
+    try {
+        left_->stop(extra);
+    } catch (const std::exception& err) {
+        *e = err;
     }
-    if (right_stop != grpc::StatusCode::OK) {
-        return right_stop;
+
+    try {
+        right_->stop(extra);
+    } catch (const std::exception& err) {
+        *e = err;
+    }
+
+    // if we received an err from either motor, throw it.
+    if (e) {
+        throw *e;
```

Comment:
Note to reviewers: the logic here should be basically the same as before except earlier we would throw the first err we encounter, and here we throw the last. This was a bit cleaner to implement and I didn't think there was a particular reason to favor one over the other, but lemme know if you disagree!